src/components/common/needsShutdown.tsx

```diff
@@ -111,6 +111,51 @@ export function needsShutdownSpice(vm: VM) {
     return vm.hasSpice !== vm.inactiveXML.hasSpice;
 }
 
+export function needsShutdownVnc(vm: VM) {
+    function find_vnc(v: VMXML): VMGraphics | undefined {
+        if (v.displays) {
+            for (const d of v.displays)
+                if (d.type == "vnc")
+                    return d;
+        }
+    }
+
+    const active_vnc = find_vnc(vm);
+    const inactive_vnc = find_vnc(vm.inactiveXML);
+
+    if (inactive_vnc) {
+        if (!active_vnc)
+            return true;
+
+        // The active_vnc.port value is the actual port allocated at
+        // machine start, it is never -1. Thus, we can't just compare
+        // inactive_vnc.port with active_vnc.port here when
+        // inactive_vnc.port is -1. Also, when inactive_vnc.port _is_
+        // -1, we can't tell whether active_vnc.port has been
+        // allocated based on some old fixed port in inactive_vnc.port
+        // (in which case we might want to shutdown and restart), or
+        // whether it was allocated dynamically (in which case we
+        // don't want to). But luckily that doesn't really matter and
+        // a shutdown would not have any useful effect anyway, so we
+        // don't have to worry that we are missing a notification for
+        // a pending shutdown.
+        //
+        if (inactive_vnc.port != "-1" && active_vnc.port != inactive_vnc.port)
+            return true;
```

Comment:
This added line is not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-2008-3f7abe69-20250716-085747-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)